src/civetweb.c

```diff
@@ -192,7 +180,25 @@ clock_gettime(int clk_id, struct timespec *t)
 	}
 	return -1; /* EINVAL - Clock ID is unknown */
 }
+
+/* if clock_gettime is declared, then __CLOCK_AVAILABILITY will be defined */
```

Comment:
So it finally always uses `_civet_clock_gettime` from [civetweb.c:150](https://github.com/civetweb/civetweb/blob/master/src/civetweb.c#L150) and ignores the OSX `clock_gettime` function that would be called [here](https://github.com/civetweb/civetweb/blob/master/src/civetweb.c#L202).

So we would need to copy this:
```C
#if !defined(_DARWIN_FEATURE_CLOCK_GETTIME) || _DARWIN_FEATURE_CLOCK_GETTIME != 0
#if __DARWIN_C_LEVEL >= 199309L
#if __has_feature(enumerator_attributes)
#define __CLOCK_AVAILABILITY __OSX_AVAILABLE(10.12) __IOS_AVAILABLE(10.0) __TVOS_AVAILABLE(10.0) __WATCHOS_AVAILABLE(3.0) 
#endif
#endif
#endif
```

I will continue discussing alternatives directly in the pull request (https://github.com/civetweb/civetweb/pull/358) instead of the file comment.